docs/user_guide/backend_configuration.rst

```diff
@@ -0,0 +1,284 @@
+    converter = ConverterPipe(data_interfaces=data_interfaces)
+
+    # Fetch available metadata
+    metadata = converter.get_metadata()
+
+    # Create the in-memory NWBFile object and retrieve a default configuration
+    backend="hdf5"
+
+    nwbfile = converter.create_nwbfile(metadata=metadata)
+    backend_configuration = converter.get_default_backend_configuration(
+        nwbfile=nwbfile,
+        backend=backend,
+    )
+
+    # Make any modifications to the configuration in this step, for example...
+    backend_configuration["acquisition/ElectricalSeriesAP/data"].compression_method = "Blosc"
+
+    # Configure and write the NWB file
+    nwbfile_path = "./my_nwbfile_name.nwb"
+    converter.run_conversion(
+        nwbfile_path=nwbfile_path,
+        nwbfile=nwbfile,
+        backend=backend,
+        backend_configuration=backend_configuration,
+    )
+
+.. note::
+
+    If you do not intend to make any alterations to the default configuration for the given backend type, then you can follow the classic workflow...
+
+    .. code-block:: python
+
+        converter = ConverterPipe(data_interfaces=data_interfaces)
+
+        # Fetch available metadata
+        metadata = converter.get_metadata()
+
+        # Create the in-memory NWBFile object and retrieve a default configuration
+        backend="hdf5"
+
+        # Configure and write the NWB file
+        nwbfile_path = "./my_nwbfile_name.nwb"
+        converter.run_conversion(
+            nwbfile_path=nwbfile_path,
+            nwbfile=nwbfile,
+            backend=backend,
+        )
+
+    and all datasets in the NWB file will automatically use the default configuration!
+
+
+Generic tools
+-------------
+
+A more generic approach that can apply to any customized in-memory ``pynwb.NWBFile`` object, not just those
+created from data interfaces and converters, would have the following structure...
+
+.. code-block:: python
+
+    from uuid import uuid4
+    from datetime import datetime
+
+    from dateutil import tz
+    from neuroconv.tools.nwb_helpers import make_or_load_nwbfile, get_default_backend_configuration, configure_backend
+    from pynwb import TimeSeries
+
+    nwbfile_path = "./my_nwbfile.nwb"
+
+    session_start_time = datetime(2020, 1, 1, 12, 30, 0, tzinfo=tz.gettz("US/Pacific"))
+    nwbfile = pynwb.NWBFile(
+        session_start_time=session_start_time,
+        session_description="My description...",
+        identifier=str(uuid4()),
+    )
+
+    # Add neurodata objects to the NWBFile, for example...
+    time_series = TimeSeries(
+        name="MyTimeSeries",
+        description="A time series from my experiment.",
+        unit="cm/s",
+        data=[1., 2., 3.],
+        timestamps=[0.0, 0.2, 0.4],
+    )
+    nwbfile.add_acquisition(time_series)
+
+    with make_or_load_nwbfile(
+        nwbfile_path=nwbfile_path,
+        nwbfile=nwbfile,
+        overwrite=True,
+        backend="hdf5",
+        verbose=True,
```

Comment:
is the verbose necessary, isn't it a default right now?  (asking out of curiosity, not prompting a change)

Reply:
right